### ISSUE TITLE
pytorch3d.ops.box3d_overlap computes wrong result for certain conditions

### ISSUE THREAD

**Issue by msieb1** [facebookresearch/pytorch3d#1054]
## 🐛 Bugs / Unexpected behaviors
I am using [this](https://github.com/facebookresearch/pytorch3d/blob/main/tests/test_iou_box3d.py#L134) test and modified it slightly to be 
```
ddx, ddy, ddz = 0.5, 0.5, 0
```
instead of 
```
ddx, ddy, ddz = random.random(), random.random(), random.random()
```
and now the test fails.

As one can easily verify, the resulting intersection volume should be 0.25 (the intersection is one quarter of a unit box). However, `box3d_overlap` gives 0.1627.

## Instructions To Reproduce the Issue:

1. Any changes you made (`git diff`) or code you wrote
```
import random
import torch
from pytorch3d.ops import box3d_overlap
device = "cuda"

UNIT_BOX = [
    [0, 0, 0],
    [1, 0, 0],
    [1, 1, 0],
    [0, 1, 0],
    [0, 0, 1],
    [1, 0, 1],
    [1, 1, 1],
    [0, 1, 1],
]

box1 = torch.tensor(
    UNIT_BOX,
    dtype=torch.float32,
    device=device,
)
# ddx, ddy, ddz = random.random(), random.random(), random.random()
ddx, ddy, ddz = 0.5, 0.5, 0
box2 = box1 + torch.tensor([[ddx, ddy, ddz]], device=device)
vol, _ = box3d_overlap(box1[None], box2[None])
gt_vol = torch.tensor(
        [[(1 - ddx) * (1 - ddy) * (1 - ddz)]],
        device=vol.device,
        dtype=vol.dtype,
    )
vol, gt_vol # 0.1627 != 0.2500
torch.allclose(vol, gt_vol) # will be False
```

2. The exact command(s) you ran:
Executed the code.

3. What you observed (including the full logs):
I observed that the results are wrong (should have given `0.25` as the result, but gives `0.1627`). Mind that for the original test that samples random shift values, it always passes. So it seems like I found a boundary case that makes the function fail.

**Comment by nikhilaravi** [comment]
@msieb1 thanks for pointing this out. We will try to reproduce the issue and get back to you.

**Comment by gkioxari** [comment]
@msieb1 
Are you using the most recent master for the `box3d_overlap`. There was a boundary case that I fixed recently. What version are you using?

**Comment by gkioxari** [comment]
I just confirmed and with master (well main now I guess) I get 
```python
vol = tensor([[0.2500]], device='cuda:0')
```

**Comment by msieb1** [comment]
I see. I used the following to install it (yesterday):
```
conda install -c pytorch pytorch=1.9.1 torchvision cudatoolkit=11.1
conda install -c fvcore -c iopath -c conda-forge fvcore iopath
conda install pytorch3d -c pytorch3d
```

So I guess I should've been using master/main? Or did I somehow not

**Comment by bottler** [comment]
I'm afraid the fix is newer than the latest release 0.6.1. You could use a nightly build or build yourself.

**Comment by gkioxari** [comment]
Closing this! @msieb1 let us know if anything else comes up!
